radix-engine-stores/src/hash_tree/types.rs

```diff
@@ -0,0 +1,1079 @@
+//! This file contains copy-pasted excerpts from a couple of Aptos source files
+//! which were used by the Aptos JMT implementation (see jellyfish.rs), with a
+//! couple of manual adjustments (aimed at reducing further dependency on Aptos).
+//! Each copied fragment starts with a "COPY-PASTED: from ..." comment.
+//! Each non-trivial adjustment is annotated with an "ADJUSTMENT: ..." comment.
```

Comment:
Note from author:

I left these comments behind just in case we want to (manually) re-sync something specific with the original sources. Please shout if you think this is not the right place for them.

Reply:
We will end up with our headers in all our files.

In this folder in particular, I want to be extra careful that we tag both.

Reply:
🛑 This aspect is on hold until we get new info from lawyers